jetty-ee9/jetty-ee9-servlet/src/test/java/org/eclipse/jetty/ee9/servlet/HttpChannelEventTest.java

```diff
@@ -146,30 +153,31 @@ public void onResponseContent(Request request, ByteBuffer content)
     @Test
     public void testRequestFailure() throws Exception
     {
-        start(new TestHandler());
+        start(new TestServlet());
 
         CountDownLatch latch = new CountDownLatch(2);
-        /* TODO
+
         connector.addBean(new HttpChannel.Listener()
         {
             @Override
             public void onRequestFailure(Request request, Throwable failure)
             {
+                //TODO never called
```

Comment:
The onRequestFailure and onComplete methods will never be called for a bad request in jetty-12, as that is determined in core and thus we do not know what context the bad request would have gone to.

So this test is invalid and we should update the javadoc of the listener class to say as much